### ISSUE TITLE
Liked songs

### ISSUE THREAD

**Issue by AirOne01** [oSumAtrIX/DownOnSpot#56]
**Is your feature request related to a problem? Please describe.**

AFAIK, it is not possible to download the entirety of the liked songs of an account, as Spotify does not provide the option to share it (which sucks imo).
Would It be possible to have an option to download the liked songs?

**Describe the solution you'd like**

I see two solutions:
- Making a playlist from the liked songs (potentially keeping it updated) from the client and downloading from it (could help for users that want to share their songs as well; might need need another issue)
- Fetching all the songs and downloading them directly

**Additional context**

---

Love your project, you are saving me a ton of time. Could help with PR in about two weeks when I'm back from vacation.

**Comment by oSumAtrIX** [comment]
I'd be up for collaborating. I currently have a big refactor/migration from a deprecated dependency stashed locally to fix #53, where I would have to solve merge conflicts after implementing this, though unrelated to this issue.

A new function can be added next to this to fetch the liked songs and collect them into a `Vec<Track>`:
https://github.com/oSumAtrIX/DownOnSpot/blob/main/src/spotify.rs#L116

At last the CLI would need to somehow be instructed to download liked songs.

**Comment by alkeryn** [comment]
idk if it has been added but i thought i could contribute some.
here is the spotify api reference:
https://developer.spotify.com/documentation/web-api/reference/get-users-saved-tracks

here is a basic python example i did of how you can get it, you can add the fields you want:
```python
import requests
import json

# Your access token from Spotify
access_token = 'YOUR TOKEN'

def get_liked_tracks(access_token):
    headers = {
        'Authorization': f'Bearer {access_token}',
        'Content-Type': 'application/json'
    }

    # Spotify API endpoint for fetching liked tracks
    url = 'https://api.spotify.com/v1/me/tracks?limit=50'

    liked_tracks = []

    while url:
        response = requests.get(url, headers=headers)
        if response.status_code != 200:
            print(f'Failed to fetch data: {response.status_code}')
            return

        data = response.json()
        for item in data['items']:
            track = item['track']
            track_info = {
                'track_name': track['name'],
                'artist': track['artists'][0]['name'],  # Assuming one artist
                'album': track['album']['name'],
                'added_at': item['added_at'],
                # 'href': track['href'],
                'href': track['external_urls']
            }
            liked_tracks.append(track_info)
            print(track_info)

        url = data.get('next')  # URL for the next page of items

    return liked_tracks

def main():
    tracks = get_liked_tracks(access_token)
    with open('dump.json', 'w', encoding="utf-8") as f:
        json.dump(tracks, f, ensure_ascii=False, indent=4)

if __name__ == '__main__':
    main()
```

also you don't need to get the next url if you want to do it in parallel you could just do it for each increment of 50 offset in parallel prolly.
